src/fmtnum.rs

```diff
@@ -113,7 +113,7 @@ macro_rules! fmtfloat {
 
                 let prev_prec = self.precision();
                 self.set_precision(None);
-                let out = self.str_unchecked(s.as_str());
+                let out = self.str_unchecked(s.as_str(), Alignment::Right);
```

Comment:
I'm not sure `default_align` is the way to go, because of the checking above (see line 85 concerning `self.fill() ... self.align()`.

I think the beginning of this method has to mutate the alignment to be the correct value, possibly based on the fill (in the case of int).

I'm also concerned that the tests didn't catch this. Maybe Alignment::Right and `fill` work? I can't recall.

Reply:
Ah, that's a good point. I'll add a test around default alignment and fill '0' that should error.

Reply:
In https://doc.rust-lang.org/std/fmt/#syntax, fill requires that align is explicitly set. So we'll never run across this case. I can still see about mutating it early though.

Reply:
However, I don't see any of the `=` treatment in the std::fmt docs. And it appears https://github.com/vitiral/strfmt/blob/master/src/formatter.rs#L154-L161 is kicking in for the test cases, so instead of unspecified we get `=`.

Reply:
Anyway, updated it so setting defaults happens earlier.